xbmc/utils/log.h

```diff
@@ -152,38 +152,23 @@ class CLog : public ISettingsHandler, public ISettingCallback
   template<typename... Args>
   static inline void FormatAndLogFunctionInternal(spdlog::level::level_enum level,
                                                   const char* functionName,
-                                                  const char* format,
+                                                  const std::string& format,
```

Comment:
I'm wondering if we should keep this as a `const char* format` because all methods calling this function expect `format` to be a `const char*` so we could avoid the implicit cast.

Reply:
but then `StringUtils::Format` uses `std::string` again so is there any real benefit?

Reply:
No the following `StringUtils::Format()` uses the `format` argument as a formatting argument for `"{0:s}: {1:s}"` so it will be perfect forwarded as a `const char*` to `fmt::format()`.

Reply:
You're right, I've updated it.